### ISSUE TITLE
FusorSV docker for Homo_sapiens_assembly38.fasta

### ISSUE THREAD

**Issue by MaestSi** [timothyjamesbecker/FusorSV#12]
Hello,
I re-post my issue here, instead of leaving it as a comment to a closed issue, to give it more visibility.
I tried running FusorSV on VCF files obtained using SVE callers run on hg38 BAM files (Reference: Homo_sapiens_assembly38.fasta from GATK bundle).
I would like to run it on NA12878_S10.vcf, NA12878_S11.vcf, NA12878_S18.vcf, NA12878_S35.vcf and NA12878_S4.vcf VCF files, stored in /mnt/cifs01/simone/NA12878/test_FusorSV/VCF_files/NA12878
This is the command line I used:
```
CHROMS='chr1,chr2,chr3,chr4,chr5,chr6,chr7,chr8,chr9,chr10,chr11,chr12,chr13,chr14,chr15,chr16,chr17,chr18,chr19,chr20,chr21,chr22,chrX,chrY,chrM'
EXCLUDE='1,9,13,14,17,36,38'
MNT_INPUT_DIR=/data
docker run -v /mnt/cifs01/simone/NA12878:$MNT_INPUT_DIR -it timothyjamesbecker/fusorsv FusorSV.py -r $MNT_INPUT_DIR/Homo_sapiens_assembly38.fasta \
-i $MNT_INPUT_DIR/test_FusorSV/VCF_files/ -f DEFAULT -M 0.5 \
-o $MNT_INPUT_DIR/test_FusorSV/FusorSV_docker_output -p 10 -c $CHROMS -E $EXCLUDE \
-m $MNT_INPUT_DIR/Homo_sapiens_assembly38.svmask.fasta_svmask.json
```
This is the output I got.
> no contig directory specified
> using stage id exclude list:[1, 9, 13, 14, 17, 36, 38]
> processing samples ['/data/test_FusorSV/VCF_files/NA12878']
> for chroms ['chr1', 'chr2', 'chr3', 'chr4', 'chr5', 'chr6', 'chr7', 'chr8', 'chr9', 'chr10', 'chr11', 'chr12', 'chr13', 'chr14', 'chr15', 'chr16', 'chr17', 'chr18', 'chr19', 'chr20', 'chr21', 'chr22', 'chrX', 'chrY', 'chrM']
> making a new coordinate offset json file
> merging the svmask regions
> svmask regions merged in 0.0 sec
> reading, parsing, partitioning and writing sample VCFs
> reading sample NA12878
> finished reading 0 out of 1 samples generating 0 partitions in 0.42 sec
> apply fusion model to sample inputs and generating fusorSV ouput
> starting fusorSV discovery on sample NA12878
> finished reading samples in 114.37 sec
> cluster merging tool processing samples
> completed = True
> run 0 in 297.46 sec
> ::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::

However, the output VCF files (NA12878_S-1.vcf and all_samples_genotypes.0.vcf) are empty a part from the header.
I already asked for help on SVE GitHub page, but wanted to try docker version too, to avoid versioning/dependencies issues.
Thank you very much for your help and your patience.
Simone

**Comment by timothyjamesbecker** [comment]
Without seeing your directory listing for /mnt/cifs01/simone/NA12878/data/test_FusorSV/VCF_files/ I can not say for certain, but I have a good guess:  FusorSV is designed to take a folder worth of samples (many samples) each having an individual sample folder full of VCF files so:

/analysis/sample1

should contain your S11, S17, S18 files and you specify this as:

-i /analysis/

indicating that there are samples inside you would like to process.

**Comment by MaestSi** [comment]
In /mnt/cifs01/simone/NA12878/test_FusorSV/VCF_files/ I only have subfolder NA12878, which contains the aforementioned VCF files.
I am now doing some tests on samples in meta_caller_RC1 folder. I am still not able to obtain a non empty VCF file with the docker version, while I am if I use the FusorSV version in SVE (that doesn't work on GRCh38). However, I noticed that neither human_g1k_v37_decoy.bed file nor an equivalent GRCh38 one is contained in data folder, but only json files. Should I have one, or the json file should be equivalent? If I use the docker one, should some python packages be installed, or all dependencies are self-contained?

**Comment by MaestSi** [comment]
Update: I verified that FusorSV (both docker version and SVE version) work with NA12878 data included in meta_caller_RC1, so the problem is not given by the number of samples; of course, the reference must be exactly the one used for variant calling (which is not exactly hg19 as the options would suggest, but  human_g1k_v37_decoy.fasta, which includes alt. haplotypes), and both human_g1k_v37_decoy_exclude.bed (which I found in 0.1.1 version) and human_g1k_v37_decoy_svmask.json work as -m input file.
So, I am quite convinced the issue is hg38-specific.
I tried to generate Homo_sapiens_assembly38_exclude.bed adding 'chr' prefix, removing from g1k_v37_decoy_exclude.bed all the alt chromosomes and doing a lift-over with UCSC tool, but still empty VCF file.
Since no error is given (the only "warning" is <finished reading 0 out of 1 samples generating 0 partitions>), I don't know what I could try.
I would just ask for a docker command line example that you guarantee works on hg38 reference, or for a more appropriate masking file.
Thanks
